operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -211,206 +225,248 @@ func (r *RedpandaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	if err := validateClusterParameters(rp); err != nil {
-		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonTerminalError, err.Error())
-
-		logger.Error(err, "validating cluster parameters")
-		r.EventRecorder.Eventf(rp, "Warning", redpandav1alpha2.EventSeverityError, err.Error())
-		return r.syncStatus(ctx, status, cluster)
-	}
+	reconcilers := []clusterReconciliationFn{
+		r.reconcileParameterValidation,
+		// we sync all our non pool resources first so that they're in-place
+		// prior to us scaling up our node pools
+		r.reconcileResources,
+		// next we sync up all of our pools themselves
+		r.reconcilePools,
+		// now we memoize the admin client onto the state
+		r.getAdminClient,
+		// now we ensure that we reconcile all of our decommissioning nodes
+		// TODO: Do we want to rate limit this as well given that it also calls the admin API?
+		// My thought is no since we want to be snappy with decommissioning.
+		r.reconcileDecommission,
+		// now reconcile cluster configuration
+		r.reconcileClusterConfig,
+		// finally reconcile all of our license information
+		r.reconcileLicense,
+	}
+
+	for _, reconciler := range reconcilers {
+		result, err := reconciler(ctx, state)
+		// if we have an error or an explicit requeue from one of our
+		// sub reconcilers, then just early return
+		if err != nil || result.Requeue || result.RequeueAfter > 0 {
```

Comment:
Doesn't need to be addressed now but I think we'd want to aggregate requeues (`min(requeueAfter)`) rather than short circuiting.

Reply:
At the moment this is really meant to be a step-wise reconciliation function where the only time that requeues happen is when we want to early return. I think we can reassess that, but right now it keeps the same functionality that it’s had